### DISCUSSION TITLE
Not able to open unsafe site even after using openBrowser({ignoreCertificateErrors: true})

### DISCUSSION THREAD

**Discussion by PriyomSaha** [getgauge/taiko#2463] [Development]
The site "https://e.clinique.na.us.rac-166-dec22-elb3.ncsastage.usva1.feature.elco.cloud/" has a browser popup authentication, so I tried the following code

`openBrowser({ignoreCertificateErrors: true});
goto('https://feature:bb#Yrp$!xNP6@e.clinique.na.us.rac-166-dec22-elb3.ncsastage.usva1.feature.elco.cloud/')`

**<html xmlns:v="urn:schemas-microsoft-com:vml"
xmlns:o="urn:schemas-microsoft-com:office:office"
xmlns:x="urn:schemas-microsoft-com:office:excel"
xmlns="http://www.w3.org/TR/REC-html40">

<head>

<meta name=ProgId content=Excel.Sheet>
<meta name=Generator content="Microsoft Excel 15">
<link id=Main-File rel=Main-File
href="file:///C:/Users/848913/AppData/Local/Temp/msohtmlclip1/01/clip.htm">
<link rel=File-List
href="file:///C:/Users/848913/AppData/Local/Temp/msohtmlclip1/01/clip_filelist.xml">
<style>

</style>
</head>

<body link="#0563C1" vlink="#954F72">

uname | pass
-- | --
feature | bb#Yrp$!xNP6

</body>

</html>
**

with following syntax **https://username:password@url**

But I am getting errors as Invalid URL for the above code mentioned.

Attached is the ScreenShot of the error

![image](https://user-images.githubusercontent.com/44141916/149362598-87fa0f6d-2c35-4099-9281-54a1f56a7243.png)

The extended error after using .trace
![image](https://user-images.githubusercontent.com/44141916/149362921-47137429-d6dd-46e1-b8d0-e37f25da35d1.png)

**Comment by zabil** [comment]
You need to pass the basic authorization as a [header](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Authorization)

You can do this as

```
await goto('example.com',{ headers:{'Authorization':'Basic cG9zdG1hbjpwYXNzd29y2A=='}})
``` 

Please note you will need to base64 encode your user name and password.

**Reply by PriyomSaha** [reply]
Thank you @zabil 
It worked and now I am able to navigate to the unsafe website.

**Comment by jonathangiber** [comment]
Hi @zabil,
will this still work if the authentication type is digest instead of basic? 
I mean with different headers

**Reply by zabil** [reply]
Haven't tried it, but it should work.
